pkg/apis/label/label.go

```diff
@@ -357,6 +363,16 @@ func (l Label) Monitor() Label {
 	return l.Component(TiDBMonitorVal)
 }
 
+// Prometheus assigns prometheus to component key in label
+func (l Label) Prometheus() Label {
+	return l.Component(PrometheusVal)
```

Comment:
Do not override the component label, just define another label, e.g. `app.kubernetes.io/app`.